src/libsyntax/parse/parser.rs

```diff
@@ -896,7 +896,9 @@ impl<'a> Parser<'a> {
     pub fn bump(&mut self) -> PResult<()> {
         self.last_span = self.span;
         // Stash token for error recovery (sometimes; clone is not necessarily cheap).
-        self.last_token = if self.token.is_ident() || self.token.is_path() {
+        self.last_token = if self.token.is_ident() ||
+                             self.token.is_path() ||
+                             self.token == token::Comma {
```

Comment:
I'm not sure about the performance impact of this bufferring.
I see I can modify `parse_lifetimes` to avoid this, but that will look unnatural.